### PR TITLE
Fix custom beneficiary form in pay

### LINKED ISSUE #1015: Custom Beneficiary Field in Project Pay modal will not take any input
## Summary

Title

## Environment

- **Device/OS**: [e.g. iOS]
- **Browser:** [e.g. chrome, safari]
- **Hardware wallet**: [e.g. Ledger w/ Metamask]

### PR DESCRIPTION
**Pull request by jmilldotdev**
## What does this PR do and why?

Closes #1015 

## Screenshots or screen recordings

image

form accepts input

image

beneficiary is set correctly in transaction

image

beneficiary receives tokens

image

no beneficiary set and connected wallet is beneficiary, as before.

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).

### CODE REVIEW

File: src/components/v2/V2Project/V2PayButton/V2ConfirmPayModal/index.tsx [hunk 1]
```diff
@@ -103,6 +103,9 @@ export default function V2ConfirmPayModal({
     if (!weiAmount) return
     await form.validateFields()
 
+    const beneficiaryFormVal = form.getFieldValue('beneficiary')
+    const txBeneficiary = beneficiaryFormVal ? beneficiaryFormVal : beneficiary
```
Comment (aeolianeth) [inline src/components/v2/V2Project/V2PayButton/V2ConfirmPayModal/index.tsx:107]: **question** do we need to rely on the `beneficiary` state variable anymore?

I'm wondering if we can just remove it and default to `userAddress` here instead:

```suggestion
    const txBeneficiary = beneficiaryFormVal ? beneficiaryFormVal : userAddress
```

**Comment by aeolianeth** [comment]
Thanks @jmilldotdev nice catch, just one comment from me 🙏

File: src/components/v2/V2Project/V2PayButton/V2ConfirmPayModal/index.tsx (hunk 1 above)
Comment (jmilldotdev) [reply src/components/v2/V2Project/V2PayButton/V2ConfirmPayModal/index.tsx]: Updated

**Review by aeolianeth — APPROVED** [review]
Nice @jmilldotdev ! Approving

**Pull request merged**